### PR TITLE
caddy auth and security

### PR DESCRIPTION
**Pull request by DaMandal0rian**
### **PR Type**
enhancement, security

___

### **Description**
- Implemented basic authentication using BCrypt hashed password to secure endpoints.
- Reduced maximum connections per host from 150,000 to 10 and 5 for different sections to enhance security.
- Disabled admin API and automatic HTTPS to prevent unauthorized access and manage HTTPS explicitly.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>Caddyfile</strong><dd><code>Implement basic authentication and enhance security settings</code></dd></summary>
<hr>

indexers/Caddyfile

<li>Added basic authentication block with BCrypt hashed password.<br> <li> Reduced maximum connections per host for improved security.<br> <li> Disabled admin API and automatic HTTPS for security reasons.<br>

</details>

  </td>
  <td><a href="https://github.com/autonomys/astral/pull/927/files#diff-8c93321a56d0377d04ecc3800259cf3f6d0f3dc73681ea63f4b91ac0b46f5ed2">+10/-2</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

**Comment by DaMandal0rian** [comment]
@marc-aurele-besner i won't merge this yet, as it will require any public endpoints to use the basic auth URL format.

File: indexers/Caddyfile [hunk 1]
```diff
@@ -7,6 +7,14 @@
     }
 }
 
+# Define basic auth
+(auth) {
+    basicauth {
+        # You can generate BCrypt hash using: caddy hash-password
+        admin $2a$14$A4H4GYb/HkbXKJXaib4.HObVwlGt0W.8WUihFg5U2KpTa8eODDvqK
```
Comment (marc-aurele-besner) [inline indexers/Caddyfile:14]: is this the password, if so, should it be commited?

Comment (DaMandal0rian) [reply indexers/Caddyfile:14]: @marc-aurele-besner it's the hash of the password.

**Review by marc-aurele-besner — APPROVED** [review]

**Review by marc-aurele-besner — APPROVED** [review]
LGTM

**Pull request merged**